owtf/api/handlers/auth.py

```diff
@@ -0,0 +1,230 @@
+                'exp': datetime.utcnow() + timedelta(seconds=JWT_EXP_DELTA_SECONDS)
+            }
+            jwt_token = jwt.encode(payload, JWT_SECRET, JWT_ALGORITHM)
+            data = {
+                "status": "ok",
+                "jwt-token": jwt_token.decode('utf-8')
+            }
+            self.success(data)
+
+        else:
+            error_msg = "Invalid login credentials"
+            data = {
+                "status": "fail",
+                "message": error_msg
+            }
+            self.success(data)
+
+    def set_current_user(self, user):
+        """
+            Set user cookie for the logged-in user
+            Setting HttpOnly attribute for preventing XSS attacks
+            Setting SameSite for preventing CSRF Attacks    
+        """
+
+        self.application.settings['cookie_secret'] = base64.b64encode(
+            uuid.uuid4().bytes + uuid.uuid4().bytes).decode('utf-8')
+        if user:
+            self.set_secure_cookie("user", tornado.escape.json_encode(user), secure=True, HttpOnly=True, SameSite='Lax')
+        else:
+            self.clear_cookie("user")
+
+
+class RegisterHandler(APIRequestHandler):
+    """Registers a new user when he provides email, name, password and confirm password"""
+
+    SUPPORTED_METHODS = ["POST"]
+
+    def post(self):
+        """
+            **Example request**:
+
+            .. sourcecode:: http
+
+            POST /api/v1/register/ HTTP/1.1
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "email": "test@test.com",
+                "password": "Test@34335",
+                "confirm_password": "Test@34335",
+                "name": "test"
+            }
+
+            **Example response**:
+
+            .. sourcecode:: http
+
+            **Successful registration response**
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "ok"
+                }
+            }
+
+            **Failed registration response**
+            HTTP/1.1 200 OK
+            Content-Encoding: gzip
+            Vary: Accept-Encoding
+            Content-Type: application/json; charset=UTF-8
+
+            {
+                "status": "success",
+                "data": {
+                    "status": "fail",
+                    "message": "Email already exists"
+                }
+            }
+
+        """
+        body_data = json.loads(self.request.body.decode('utf-8'))
+        name = body_data['name']
+        email = body_data['email']
+        password = body_data['password']
+        confirm_password = body_data['confirm_password']
+
+        already_taken = User.find_by_email(self.session, email)
+        reg = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!#%*?&]{8,20}$"
```

Comment:
do you have a test that this regex works as expected?

Reply:
Moved this password check regex in settings.py with other such regexes. Also, added another regex for email validation.
![image](https://user-images.githubusercontent.com/37950602/121951139-29716900-cd78-11eb-8c29-52f74d43227a.png)

Password validation checks whether the password is strong enough by checking:
- small Letters[a-z]
- Capital letters[A-Z]
- digit[0-9]
- Special Characters(Except Space)
- Atleast 8 chars and almost 20 chars.

Shown below are a few examples I run against this regex at this platform https://regex101.com/.
![image](https://user-images.githubusercontent.com/37950602/121950964-f16a2600-cd77-11eb-9187-9e190433499b.png)

Similar test results for email validation regex:
![image](https://user-images.githubusercontent.com/37950602/121952175-84f02680-cd79-11eb-99db-301c7c609c37.png)